src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -255,10 +255,10 @@ private static TagHelperAttributeDescriptor ToAttributeDescriptor(PropertyInfo p
 
         private static bool IsAccessibleProperty(PropertyInfo property)
         {
-            return property.GetMethod != null &&
-                   property.GetMethod.IsPublic &&
-                   property.SetMethod != null &&
-                   property.SetMethod.IsPublic;
+            // Accessible properties are those with public getters and setters and without [HtmlAttributeNotBound].
+            return property.GetMethod?.IsPublic == true &&
+                property.SetMethod?.IsPublic == true &&
+                property.GetCustomAttribute<HtmlAttributeNotBoundAttribute>(inherit: false) == null;
```

Comment:
> why are we ignoring inheritance entirely?

I asked the same question ages ago. The decision was to have all attributes `sealed` with `Inherited = false`. IIRC the intent was to require explicit attributes in a subclass of a tag helper.

In this particular case, note properties inherited from a tag helper bring along their attributes.  `Inherited` primarily matters for properties when a `virtual` property has an `override` in the subclass. See `TagHelperDescriptorFactoryTest` for some examples.